### ISSUE TITLE
Use effect: should pass wrapped target as argument?

### ISSUE THREAD

**Issue by dy** [spectjs/spect#63]
From modular effects standpoint, that tightly couples `use` with `$`.
Imagine we'd like to use atomic `use` as:

```js
import use from 'spect/use'
import state from 'spect/state'

let items = [a, b, c]

use.call(items, (item) => {
  state.call(item, {x: 1}, [])
  
  console.log(state.call(item, 'x'))
  setTimeout(() => {
    state.call(item, s => s.x++)
  }, 1000)
})
```

`$` can be useful here, but redundant in general:

```js
import $ from 'spect'

let items = $([a, b, c])

items.use(item => {
  item.state({x: 1}, [])
  
  console.log(item.state.x)

  setTimeout(() => {
    item.state(s => s.x++)
  }, 1000)
})
```

jQuery is also in general against that, it doesn't even pass item as attribute to `each`, and passes index instead:
```js
$items.each(i => {
let $item = $(this)
})
```

So due to these reasons - tight coupling `use` with `$` and incompatibility with `jQuery` there's a point in passing direct element to effect:

```js
$items.use(el => {
$(el).html
})
```

That would indeed enable destructuring props as well #56 #48 

```js
$items.use( ({href, prop}) => {
})
```

And be compatible with standard `forEach(el => {})`.

:star: So, the result: pass unwrapped element to `use` to be modular, forEach/jquery-complacent, destructure-able.

**Comment by dy** [comment]
See #86.
The main issue of passing direct element is conceptual mess:
```js
const App = el => $(el).html`${ data.map(item => $`<${ Card } ...${item}/>`) }`
// vs
const App = ({ html, prop }) => html`${ prop('data').map(item => html`<${ Card } ...${item} />` ) }`
```
If we keep domain accessor, we lose atomic effects argument, which is nice too.
But having 
```js
const App = (el) => {
  let _html = html.bind(el)
  let _fx = _fx.bind(el)
}
```
is tedious. That's why $ way with connected effects has pros.

The vue3 approach, proposed by @vuhg is passing both el/props and wrapped el:
```js
const App = ({ data }, { html }) => html`${ data.map(item => html`<${ Card } ...${ item }/>`) }`
// which is essentially the same as
const App = el => { let { html } = $(el); return html`${ data.map( item => html`...` ) }` }
```

**Comment by dy** [comment]
Fixed in #86.
